core/src/Streamly/Internal/Data/Parser/ParserD.hs

```diff
@@ -1774,12 +1842,19 @@ wordWithQuotesWith isInvalid keepQuotes isEsc toRight isSep
                    then if keepQuotes
                         then processUnquoted s a
                         else return $ Continue 0 $ WordUnquotedWord s
-                   else process s a (n - 1) ql qr
+                   else processQuoted s a (n - 1) ql qr
                 else if a == ql
-                     then process s a (n + 1) ql qr
-                     else process s a n ql qr
+                     then processQuoted s a (n + 1) ql qr
+                     else processQuoted s a n ql qr
     step (WordUnquotedEsc s) a = processUnquoted s a
-    step (WordQuotedEsc s n ql qr) a = process s a n ql qr
+    step (WordQuotedEsc s n ql qr) a =
+        case tr ql a of
+            Nothing -> do
+                res <- fstep s escChar
+                case res of
+                    FL.Partial s1 -> processQuoted s1 a n ql qr
+                    FL.Done b -> return $ Done 0 b
```

Comment:
If there is no translated char you keep the escape char and the char after it.
Maybe you should error out here?
Shouldn't escape char always be processed?
You should never escape something that is not escapable.

Reply:
This is actually pretty general.
You can escape the quotes or escape the escape character.

Reply:
For example,
`"abc\"\\d"` is valid.
`"abc\"\d"` should be invalid.
The function above can be used to represent both I guess.

Reply:
NVM, actually looks good.